web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -152,8 +152,26 @@ def add_file_record(
         return file_record.id
 
     try:
-        file_record = File(file_path, content_hash, None, None)
-        session.add(file_record)
+        # Parallel storage of runs containing common file paths results a
+        # "duplicate key violation" error. This is handled by CodeChecker, so
+        # practically it causes no problem. The INSERT command of the second
+        # transaction will be thrown away. However, some DB systems are
+        # supporting "ON CONFLICT DO NOTHING" clause in INSERT statement which
+        # solves the same issue gracefully.
+        # TODO: "ON CONFLICT DO NOTHING" feature is available for SQLite engine
+        # too in SQLAlchemy 1.4.
+        if session.bind.dialect.name == 'postgresql':
+            insert_stmt = sqlalchemy.dialects.postgresql.insert(File).values(
+                filepath=file_path,
+                filename=os.path.basename(file_path),
+                content_hash=content_hash).on_conflict_do_nothing(
+                    index_elements=['id'])
```

Comment:
index_elements=["content_hash"]?? Shouldn't we list the key clashing constraint fields here?

Reply:
Maybe you can test this fix and provoke the error by storing a file to the server (with masstore run) that has been stored already.

Reply:
[2022-08-15 09:30:01 UTC] <???> (23505) - ERROR:  duplicate key value violates unique constraint "uq_files_filepath"
[2022-08-15 09:30:01 UTC] <???> (23505) - DETAIL:  Key (filepath, content_hash)=(/path/to/blabla.h, <hash_value>) already exists.
[2022-08-15 09:30:01 UTC] <???> (23505) - STATEMENT:  INSERT INTO files (filepath, filename, content_hash, remote_url, tracking_branch) VALUES ('/path/to/blabla.h', 'blabla.h', '<hash_value>', NULL, NULL) RETURNING files.id